bin/server/cmd/root.go

```diff
@@ -232,162 +213,34 @@ func start() {
 	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
 	go func() {
 		<-c
-		m.l.Info("SIGINT received.")
-		if err := m.Close(); err != nil {
-			fmt.Fprintln(os.Stderr, err)
+		logger.Info("SIGINT received.")
+		if s != nil {
+			_ = s.Shutdown()
 		}
 		cancel()
 	}()
 
-	// Execute program.
-	if err := m.Run(ctx); err != nil {
-		if err != http.ErrServerClosed {
-			m.l.Error(err.Error())
-			m.Close()
-			cancel()
-		}
-	}
-
-	// Wait for CTRL-C.
-	<-ctx.Done()
-}
-
-// NewMain creates a main server based on profile.
-func NewMain(activeProfile server.Profile, logger *zap.Logger) (*Main, error) {
-	fmt.Println("-----Config BEGIN-----")
-	fmt.Printf("mode=%s\n", activeProfile.Mode)
-	fmt.Printf("server=%s:%d\n", host, activeProfile.Port)
-	fmt.Printf("datastore=%s:%d\n", host, activeProfile.DatastorePort)
-	fmt.Printf("frontend=%s:%d\n", frontendHost, frontendPort)
-	fmt.Printf("demoDataDir=%s\n", activeProfile.DemoDataDir)
-	fmt.Printf("readonly=%t\n", readonly)
-	fmt.Printf("demo=%t\n", demo)
-	fmt.Printf("debug=%t\n", debug)
-	fmt.Println("-----Config END-------")
-
-	metadataDB, err := createMetadataDB(&activeProfile, logger)
+	s, err = server.NewServer(ctx, &activeProfile, logger, level)
```

Comment:
Can we print greeting banner somewhere here so we don't need to pass that to the profile?

Reply:
I think it can also be considered, `NewServer()` handles a lot of things, and it would be weird to print Banner in the first place.

Reply:
> I think it can also be considered, `NewServer()` handles a lot of things, and it would be weird to print Banner in the first place.

@tianzhou Oh, I get your idea. That sounds good. But in my assumption, our tests only call `NewServer()`, the tests may not observe the greeting banner if we print the greeting banner in `start()`.